packages/xstate-graph/src/shortestPaths.ts

```diff
@@ -133,8 +132,40 @@ export function getShortestPaths<
   );
 
   if (resolvedOptions.toState) {
-    return paths.filter((path) => resolvedOptions.toState!(path.state));
+    return paths
+      .filter((path) => resolvedOptions.toState!(path.state))
+      .map(alterPath);
   }
 
-  return paths;
+  return paths.map(alterPath);
+}
+
+export function alterPath<T extends StatePath<any, any>>(path: T): T {
+  let steps: T['steps'] = [];
+
+  if (!path.steps.length) {
```

Comment:
it feels off that we need to alter the steps/paths after the fact, couldn't all of this be just made upfront? in other words: why the internal shape of this is different than the public one that we return to the user?

Reply:
Comment added for this (thanks)